Remove sim.sh, insert contents into .bashrc in Dockerfile

amarburg opened the pull request:
## Changes Made

The environment variable "GZ_VERSION" for the Gazebo version is set in the Dockerfile.   Don't set it in "sim.sh"

## Testing

In progress.

evan-palmer commented:
Thanks for adding this. I've been considering getting rid of the `sim.sh` file altogether because it is written primarily for the Docker image. I originally added it to help with source installation, but the docs [encourage](https://github.com/Robotic-Decision-Making-Lab/blue/blob/main/docs/docs/installation.mdx#source-installation) using the Dockerfile as a reference for installation, so it doesn't make much sense to keep it around

amarburg commented:
I originally made the change that way (moved all the "export"s to Dockerfile) but thought I might be missing a use case.    

As you say, I think it disadvantages installing from source on bare metal -- you need to manually set up those GZ_* env variables.    But I think that sort of bare metal install would require more focused attention and documentation to run smoothly in any case.

amarburg commented:
Removed sim.sh, moved contents into .bashrc through the Dockerfile

(since PATH, GZ_SIM_RESOURCE_PATH, and GZ_SIM_RESOURCE_PLUGIN modify existing environment variables, they need to be evaluated at runtime in .bashrc, can't be set by an ENV)

evan-palmer commented on .docker/Dockerfile:
```diff
@@ -192,7 +192,26 @@ RUN . "/opt/ros/${ROS_DISTRO}/setup.sh" \
+export GZ_SIM_RESOURCE_PATH=\$HOME/ardupilot_gazebo/models:\$HOME/ardupilot_gazebo/worlds:\$GZ_SIM_RESOURCE_PATH
+
+# Add blue models and worlds
+export GZ_SIM_RESOURCE_PATH=\$HOME/ws_blue/src/blue/blue_description/gazebo/models:\$HOME/ws_blue/src/blue/blue_description/gazebo/worlds:\$GZ_SIM_RESOURCE_PATH
+
+# Add blue meshes
+export GZ_SIM_RESOURCE_PATH=\$HOME/ws_blue/src/blue/blue_description/meshes:\$GZ_SIM_RESOURCE_PATH
```
Does `${USER_WORKSPACE}` work here? If so, we can shorten this a bit using that variable.

amarburg replied:
Added an intermediate variable (in shell, not in Docker) `$BLUE_DESCRIPTION` which points to `.../blue_description/`

evan-palmer reviewed (approved):
Looks good! Thanks for making these changes.

evan-palmer commented:
@mergify backport jazzy iron humble
